pkg/command/applications.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2021 GramLabs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/thestormforge/optimize-go/pkg/api"
+	applications "github.com/thestormforge/optimize-go/pkg/api/applications/v2"
+)
+
+func newApplicationsCommand(cfg Config) *cobra.Command {
+	return &cobra.Command{
+		Use:               "applications [NAME ...]",
+		Aliases:           []string{"application", "apps", "app"},
+		ValidArgsFunction: validApplicationArgs(cfg),
+	}
+}
+
+// NewGetApplicationsCommand returns a command for getting applications.
+func NewGetApplicationsCommand(cfg Config, p Printer) *cobra.Command {
+	var (
+		batchSize int
+	)
+
+	cmd := newApplicationsCommand(cfg)
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		ctx, out := cmd.Context(), cmd.OutOrStdout()
+		client, err := api.NewClient(cfg.Address(), nil)
+		if err != nil {
+			return err
+		}
+
+		l := applications.Lister{
+			API:       applications.NewAPI(client),
+			BatchSize: batchSize,
+		}
+
+		result := &ApplicationOutput{Items: make([]ApplicationRow, 0, len(args))}
+		if err := l.ForEachNamedApplication(ctx, args, false, result.Add); err != nil {
+			return err
+		}
+
+		if len(args) == 0 {
+			q := applications.ApplicationListQuery{}
+			if err := l.ForEachApplication(ctx, q, result.Add); err != nil {
+				return err
+			}
+		}
```

Comment:
nit, this reads a little odd. 
it partially makes me want to have `ForEachNamedApplication` be renamed to `ForEachApplication` and then treat len(args)==0 special internally there. not an ask for change exactly, just thinking out loud

Reply:
I agree it reads a little strange. I toyed with the idea of adding names to the `ApplicationListQuery` but it breaks the model of have having it map to the query string. I also thought about consolidating the functions (ForEach and ForEachNamed), but the logic was different enough that it seem like a bad idea.

Would this be a readability win here?
```
switch len(args) {
case 0:
  // ForEachApplication
default:
  // ForEachNamedApplication
}
```

Reply:
Or...gasp...an `if/else`? There isn't ever an argument for anything other then the two cases...